### PR TITLE
Add TestCommon project

### PR DESCRIPTION
**Pull request by rynowak**
Adds a TestCommon project for shared files used in our unit tests. Moves `TestModelMetadataProvider` to it.

### CODE REVIEW

**Comment by pranavkm** [comment]
Could you also move the `TestFileProvider` from `Razor.Host` to the common project?

File: test/Microsoft.AspNet.Mvc.TestCommon/Microsoft.AspNet.Mvc.TestCommon.xproj [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="__ToolsVersion__" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.TestCommon/Microsoft.AspNet.Mvc.TestCommon.xproj]: run script to fix .xproj files prior to push

File: test/Microsoft.AspNet.Mvc.TestCommon/TestModelMetadataProvider.cs [hunk 2]
```diff
@@ -63,7 +62,7 @@ public IMetadataBuilder ForProperty(Type containerType, string propertyName)
             var property = containerType.GetRuntimeProperty(propertyName);
             Assert.NotNull(property);
 
-            var key = ModelMetadataIdentity.ForProperty(property.PropertyType, propertyName, containerType);
+            var key = ModelMetadataIdentity.ForProperty(property.PropertyType, property.Name, containerType);
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.TestCommon/TestModelMetadataProvider.cs]: curious: when would the `propertyName` value be different from `property.Name`?
and why ditch the parameter in favour of the property?

**Comment by dougbu** [comment]
:shipit: once @pranavkm is mildly pleased

File: test/Microsoft.AspNet.Mvc.TestCommon/TestModelMetadataProvider.cs (hunk 2 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.TestCommon/TestModelMetadataProvider.cs]: No idea, I copy-pasted this file.

**Pull request closed without merging**